AAT_gae/src/main/java/com/aat/restlet/AttendanceResource.java

```diff
@@ -19,17 +19,24 @@
 import com.aat.datastore.Course;
 import com.aat.utils.ResourceUtil;
 
+import java.util.List;
+
 public class AttendanceResource extends ServerResource {
 	
 	@Post
 	public Representation create(Representation entity) {
+		// Check if of type Student
+		ResourceUtil.checkTokenPermissions(this, Student.class);
```

Comment:
As now all of the resources (maybe except signin and signup) need to do checkTokenPermissions, maybe we could have a base class for them and do this common calls there?